src/main/java/org/truffleruby/language/globals/ReadMatchReferenceNodes.java

```diff
@@ -128,7 +117,7 @@ public Object execute(VirtualFrame frame) {
 
             final DynamicObject matchData = (DynamicObject) match;
 
-            return tosMatchNode.execute(matchData);
+            return getMatchIndexNode.getIndex(matchData, 0, NotProvided.INSTANCE, normalizedIndexProfile, indexOutOfBoundsProfile, hasValueProfile);
```

Comment:
Definite use case for a abstract execute() method, this is breaking encapsulation.